File: go/two-fer/two_fer.go
// Package twofer is used to implement the exercism rules for sharing.
package twofer

import "fmt"

// ShareWith will share with name if one is passed in, otherwise the pronoun "you" will be used.
func ShareWith(name string) string {
	if len(name) == 0 {
		return "One for you, one for me."
	}
	
	return fmt.Sprintf("One for %v, one for me.", name)
}

File: csharp/two-fer/TwoFer.cs
using System;

public static class TwoFer
{
    public static string Speak(string name=null)
    {
      return name == null ? respond("you") : respond(name);
    }
    private static string respond(string name)
    {
      return $"One for {name}, one for me.";
    }
}
